.github/workflows/create-deploy-release.yml

```diff
@@ -0,0 +1,79 @@
+name: Create and deploy a new release
+run-name: create and deploy ${{ github.sha }}
+on:
+  push:
+    branches:
+      - create-deploy-release # TODO: switch to main branch before merge
+jobs:
+  trigger-gitlab-pipeline:
+    if: startsWith(github.event.head_commit.message, '[') # regexp not supported
```

Comment:
Si on reste sur un mécanisme par commit, comment faire pour déployer : 
1. HEAD si on a pas mis le trigger dans le message de commit ?
2. Un commit arbitraire de l'historique pour rollback ?

Reply:
1. you could do a simple empty "deploy" commit:
```
$  git commit --allow-empty -m "[demo:ecospheres:minor] trigger deployment"
```
2. either replay the previous workflow or if there is no workflow associated with the commit, create one (checkout a branch, revert to this commit, etc...)

Reply:
Ok donc c'est faisable mais IMO moins pratique que par tag. Du coup j'ai du mal à voir l'intérêt de passer par commit msg plutôt que tag ?

Reply:
We prefer to centralize and standardize the way we're bumping, tagging and building images, and to let the CI/CD do that: this is done via our scaffolding scripts which is a single repository used by all our apps (in that way we could easily update and make things available for all our application repositories).

Reply:
Les tags ne se font pas au même endroit, si ?
- Le tag dont je parle se fait sur udata-front-kit, et on pourrait imaginer que différentes $app utilisent différents tags.
- Le tag dont tu parles se fait côté infra data.gouv (CICD et image docker), j'imagine via `./scripts/gitlab-ci-pipeline.sh` ?

En gros on a : 
1. Mécanisme de trigger CICD. User-facing, faire au plus simple/flexible d'utilisation.
2. Tag CICD/image. Votre popote, boite noire vu d'ici.
3. Relier les deux. Idem dans le cas d'un tag ou d'un commit message en step 1, dans tous les cas on parle d'une ref git.

Pour phase 1, je pense que tag >= commitmsg. Dans les deux cas vous pouvez extraire l'info nécessaire à partir du pattern du tag/commitmsg et l'utiliser de la même façon pour définir votre tag en phase 2. Ca ne change donc pas grand chose au final, si ce n'est que côté user ça simplifie les cas de deploiement a posteriori.

Reply:
Using commit messages instead of direct tagging allows us to standardize the way the tags are named.

Also, we do want a **strong relationship** between repository and image tags, and even between apps/repositories. We do not want to allow custom tags naming.  Apart from an extra commit message, it doesn't make much difference for the developer: you will ends up with a tag anyway (and I agree with you : tags > commit messages).

The commit message is just here to handle git tagging for you, for the reason I exposed above. Tags creation should be handled by the workflow itself.

Reply:
Seems I misunderstood where the tagging occurred, sorry. I thought you tagged somewhere on your side (it'd help to see `./scripts/gitlab-ci-pipeline.sh` for the review btw).

If the tagging happens in this repo only, I get the need to match images with sources (and agree). We could still have a tagging pattern to trigger deploys and another one for code/image mapping, with both tags ending up on the same commit ref.

I still think we'd get more from trigger-by-tag: 
- I don't think you can make an empty commit from GitHub, and would prefer a way to trigger deploys from both UI and CLI.
- Relying on commit means anyone with commit rights can deploy. With tags we can further restrict who will be able to deploy. Possibly even handle it by environment (not sure if the [tag protection rules](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-tag-protection-rules) are precise enough for that though).
- I'm not sure semver is going to be easy with multiple sites deploying from the same repo. We're not really packaging "proper releases" that we then deploy, but rather just deploy commit refs. Semver would require careful analysis of the [upcoming changelog](https://github.com/ecolabdata/ecospheres/issues/196) each time someone deploys their site to decide what version number to bump. For ecologie.dgfr, we thought of simply going with a date+number (eg 20240423-1) for versions to avoid that question, at least for now.
- Redeploying still seems easier adding a new tag than running through the workflow list to find the one that needs retriggering (is there a way to find a given run from the tag added within that run?)

Reply:
Dans le cas d'un deploy-by-commit : 
Je commit-deploy "[demo:ecospheres:...]" sur la branche `ecospheres-demo` -> deploy sur l'env demo, normal.
Je merge la branche `ecospheres-demo` sur `ecospheres-prod`, ce qui inclut le commit-deploy "[demo:ecospheres:...]".
Est-ce que le workflow va re-trigger sur la branche `ecospheres-prod` et donc déployer la branche prod sur l'env demo ?